static/app/views/releases/utils/useFetchThresholdsListData.tsx

```diff
@@ -0,0 +1,74 @@
+import {useCallback, useEffect, useState} from 'react';
+
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+
+import {Threshold, ThresholdQuery} from './types';
+
+export const EMPTY_THRESHOLDS_LIST_DATA: ReturnType<typeof useFetchThresholdsListData> = {
+  errors: null,
+  isLoading: false,
+  thresholds: [],
+};
+
+export type HookProps = {
+  selectedEnvs: string[];
+  selectedProjects: number[];
+};
+
+export default function useFetchThresholdsListData({
+  selectedEnvs,
+  selectedProjects,
+}: HookProps) {
+  const api = useApi();
+  const organization = useOrganization();
+
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [errors, setErrors] = useState<string | null>();
+  const [_featureEnabledFlag, setFeatureEnabledFlag] = useState<boolean>(true);
+  const [thresholds, setThresholds] = useState<Threshold[]>([]);
+
+  const fetchThresholds = useCallback(async () => {
+    //     ReleaseThresholdIndexEndpoint.as_view(),
+    //     name="sentry-api-0-organization-release-thresholds",
+    const path = `/organizations/${organization.id}/releases/thresholds/`;
+    const query: ThresholdQuery = {};
+    if (selectedProjects.length) {
+      query.project = selectedProjects;
+    } else {
```

Comment:
this should be using useApiQuery unless there's some reason that this can't use it

Reply:
🤔 
hmmm. why's that?

Seems like `useApiQuery` is just another abstraction on top of `useApi`.

it unnecessarily obfuscates the implementation here?

Reply:
useApiQuery is a small wrapper on react-query that uses our api client to fetch but it handles loading states, error states, stale time, refetching based on query parameters. basically everything you're doing here

Reply:
hmmm... i see 🤔 
Ok that looks useful. will switch

Reply:
👍  might be helpful to start by looking at one like this one https://github.com/getsentry/sentry/blob/master/static/app/actionCreators/group.tsx#L518-L545